src/Core/Components/NavMenu/FluentNavMenu.razor.js

```diff
@@ -22,6 +22,18 @@ export function onUpdate() {
     
 }
 
+export function onDispose() {
+    for (let expander of document.getElementsByClassName("expander")) {
+        if (expander) {
+            expander.removeEventListener('click', toggleMenuExpandedAsync);
+            expander.removeEventListener('keydown', handleMenuExpanderKeyDownAsync);
+        }
+    }
+    for (let element of document.getElementsByClassName("fluent-nav-group")) {
+        detachEventHandlers(element);
+    }
+}
+
 function attachEventHandlers(element) {
     let navlink = element.getElementsByClassName("fluent-nav-link")[0];
     if (!navlink.href) {
```

Comment:
Perhaps you should add a test to check whether the `navLink` element has been found (to handle all cases)